sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -32,9 +32,7 @@ object NestedColumnAliasing {
 
   def unapply(plan: LogicalPlan)
     : Option[(Map[GetStructField, Alias], Map[ExprId, Seq[Alias]])] = plan match {
-    case Project(projectList, child)
-        if SQLConf.get.nestedSchemaPruningEnabled && canProjectPushThrough(child) =>
-      getAliasSubMap(projectList)
+    case Project(projectList, child) => getAliasSubMap(projectList)
```

Comment:
@viirya . Sorry, but this is a regression on all the existing code. We should avoid `getAliasSubMap` invocation. https://github.com/apache/spark/pull/24637/files#diff-a636a87d8843eeccca90140be91d4fafR635 doesn't prevent `getAliasSubMap` invocation inside `unapply`.

Reply:
I see. If so, I need to make a little change to prevent it. Will change it later.

Reply:
Thanks!